### DISCUSSION TITLE
How can i use it with github actions?

### DISCUSSION THREAD

**Discussion by harisvsulaiman** [basecamp/kamal#188] [Q&A] [answered]
I need to use mrsk to deploy apps using github actions. How can i share the ssh key for this to work?

**Comment by jeremy** [comment, marked as answer]
Generate a SSH key for deploys to your servers and add it to GitHub as a secret. Then use a GitHub action to install the SSH key before running MRSK.

You can automate some of this with prebuilt actions such as these:
* https://github.com/marketplace/actions/install-ssh-key
* https://github.com/marketplace/actions/webfactory-ssh-agent

Good luck! Please do share your results.

**Reply by northeastprince** [reply]
@jeremy does 37signals do it this way? I thought it was done locally but that seems to just add risk and a higher change of liability, as well as having to manually deploy on each change.

**Comment by huksley** [comment]
Also, you might need to create a separate GitHub token as the default one does not give access to the ghcr.io docker repository.

**Reply by northeastprince** [reply]
You can use the default token secret injected into every Action and just make sure it's set to allow pushing/pulling Packages.

**Comment by harisvsulaiman** [comment]
@jeremy Please consider adding this to the docs
